variant-inc/terraform-aws-s3
# Feature/cloud 911 s3

# Description

Created s3 module with ignore changes to replication rules.

Fixes [#911](https://drivevariant.atlassian.net/browse/CLOUD-911https://usxtech.atlassian.net/browse/CLOUD-8)

## Type of change


## How Has This Been Tested?

```yaml
Pre reqs:
  - Create terraform.tfvars.json (provided in readme.md)
  - Create provider.tf (provided in readme.md)
  - In version.tf, change (backend s3) -> (backend local)
  - Clone/Pull branch
  - Export DPL Account variables
 
Test Case 1 [Create S3 bucket locally]:
- Steps
  - Terraform validate
  - Terraform init
  - Terraform plan/apply
- Result
 - See Output Bucket name and Bucket arn in terminal
 - See Bucket Created in dpl

Test Case 2 [Destroy S3 Bucket locally]:
- Steps
  - terraform destroy
- Results
  - See Bucket deleted

Test Case 3 [Create S3 Bucket with backend s3]:
- Steps
  - In version.tf, change (backend local) -> (backend s3)
  - remove statefiles/lock/.terraform
  - terraform init \
      -backend-config="bucket=test-bucket-156gpjfbx4m63stm" \
      -backend-config="key=bucket/test" \
      -backend-config="region=us-west-2" \
      -backend-config="dynamodb_table=lazy_tf_state" \
      -backend-config="encrypt=true"
  - terraform plan/apply
- Results
 - See Output Bucket name and Bucket arn in terminal
 - See Bucket Created in dpl
 - See State Key for CREATED BUCKET in S3 Bucket test-bucket-156gpjfbx4m63stm
 - See State Lock for CREATED BUCKET in lazy_tf_state Dynamo DB


Test Case 4 [Delete S3 Bucket with backend s3]:
- Steps
  - terraform destroy
- Results
  - See Bucket deleted
```

## Checklist
